Library/Homebrew/dev-cmd/tests.rb

```diff
@@ -80,6 +80,10 @@ def tests
         files = files.reject { |p| p =~ %r{^test/(os/mac|cask)(/.*|_spec\.rb)$} }
       end
 
+      unless OS.linux?
+        files = files.reject { |p| p =~ %r{^test/os/linux(/.*|_spec\.rb)$} }
```

Comment:
This prevents the Linux-specific tests from running on (and consequently breaking) macOS.